standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -2435,8 +2435,24 @@ public Type getType(String name) throws NoSuchObjectException, MetaException, TE
   @Override
   public List<String> getTables(String catName, String dbName, String tablePattern)
       throws TException {
-    List<String> tables = client.get_tables(prependCatalogToDbName(catName, dbName, conf), tablePattern);
-    return FilterUtils.filterTableNamesIfEnabled(isClientFilterEnabled, filterHook, catName, dbName, tables);
+    List<String> tables = new ArrayList<>();
+    GetProjectionsSpec projectionsSpec = new GetProjectionsSpec();
+    List<String> projectedFields = Arrays.asList("dbName", "tableName", "owner", "ownerType");
```

Comment:
We should use GetTableProjectionSpecBuilder instead of hard coding the fieldnames. builder.includeDatabase().includeTableName().includeOwner().includeOwnerType()

Reply:
GetTableProjectionsSpecBuilder is not importable in HiveMetaStoreClient class so we'll have to do this in this way only I guess.